### PR TITLE
Add benchmark script and workflow job

### PR DESCRIPTION
**Pull request by vardbabayan**

### CODE REVIEW

**Comment by 0crat** [comment]
@vardbabayan Hey there! 👋 Just a heads up about your Git branch naming. It looks like "`vb/factbase_benchmark`" doesn't quite follow our convention. Remember, [it's best](https://www.yegor256.com/2014/04/15/github-guidelines.html) to name branches after the ticket number you're working on. For example, "`131`" would be perfect here. Our policy deducts 12 points for non-standard names, so let's avoid that next time. Keep up the good work, and let's stick to the guidelines to keep our repo tidy! 🚀

**Comment by yegor256** [comment]
@vardbabayan thanks for the contribution! Please, check that all CI jobs complete successfully (at the moment, they don't)

**Comment by yegor256** [comment]
@vardbabayan at the moment, the `benchmark_factbase.rb` is not part of `rake` build, which makes it impossible to check its validity during the normal build. I suggest adding it to `Rakefile`, something like this:

```
Rake::TestTask.new(:test) do |test|
  Rake::Cleaner.cleanup_files(['coverage'])
  test.libs << 'lib' << 'test' << 'benchmark' # here!
end

```

**Comment by vardbabayan** [comment]
Added benchmark test as part of rake build in ci and updated yamllint configuration

**Comment by yegor256** [comment]
@vardbabayan looks good, but there is no `<!-- benchmark_begin -->` in the `README.md` -- nothing to replace

**Comment by vardbabayan** [comment]
@yegor256 Added missed comment in the `README.md`

File: .github/workflows/benchmark.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            printf "\`\`\`\n\n"
+            echo "The results were calculated in [this GHA job][benchmark-gha]"
+            echo "on $(date +'%Y-%m-%d') at $(date +'%H:%M'),"
+            echo "on $(uname) with $(nproc --all) CPUs."
+          )
+          export sum
+          perl -i -0777 -pe 's/(?<=<!-- benchmark_begin -->).*(?=<!-- benchmark_end -->)/\n$ENV{sum}\n/gs;' README.md
+          url=${GITHUB_SERVER_URL}/${GITHUB_REPOSITORY}/actions/runs/${GITHUB_RUN_ID}
+          export url
+          perl -i -0777 -pe 's/(?<=\[benchmark-gha\]: )[^\n]+(?=\n)/$ENV{url}/gs;' README.md
```
Comment (yegor256) [inline .github/workflows/benchmark.yml:67]: @vardbabayan what this command will be doing with the README?

Comment (vardbabayan) [reply .github/workflows/benchmark.yml]: @yegor256 it sets up a link to the github action that has executed the benchmark, probably the link is redundant, so I removed it

Comment (vardbabayan) [reply .github/workflows/benchmark.yml:67]: @yegor256 I see the link is in between  and  comments, [it's here](https://github.com/yegor256/factbase/pull/132/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R171)

Comment (yegor256) [reply .github/workflows/benchmark.yml:67]: @vardbabayan but you're searching for `benchmark-gha`, which doesn't exist in README

Comment (vardbabayan) [reply .github/workflows/benchmark.yml:67]: @yegor256 placeholder `benchmark-gha` has been added, linter passed
let's merge?

**Comment by yegor256** [comment]
@vardbabayan it seems that benchmark(s) are not tests and not executed by minitest. It's better to place them in the `benchmark/` directory, instead of `test/benchmark/`.

**Comment by 0crat** [comment]
@vardbabayan Thank you for your contribution! We appreciate your effort, but there are a few areas for improvement. You've earned +4 points: +4 as a base, -8 for exceeding the recommended hits-of-code (221 ≥ 200), and -16 for the absence of a code review. However, we've adjusted it to +4 to maintain the minimum reward. Please remember to keep your contributions [concise](https://www.yegor256.com/2018/03/06/speed-vs-quality.html) and ensure code reviews are conducted. Your current balance stands at [-24](https://zerocracy.github.io/judges-action/zerocracy-vitals.html). We look forward to your future contributions!

**Pull request merged**